### PR TITLE
SW-2344 / SW-2343 / SW-2345 / SW-2331 Misc fixes around showing/hiding withdraw buttons when relevant

### PR DESCRIPTION
**Pull request by karthikbtf**
- also fixed add batch entry form remembering entries
Terraware App 2022-11-22 09-41-43
Terraware App 2022-11-22 09-40-55
Terraware App 2022-11-22 09-27-46
Terraware App 2022-11-22 09-27-29
Terraware App 2022-11-22 09-27-00
Screen Shot 2022-11-22 at 9 55 22 AM
Terraware App 2022-11-22 10-05-15
Terraware App 2022-11-22 10-04-50

### CODE REVIEW

File: src/components/Inventory/view/InventorySeedlingsTable.tsx [hunk 1]
```diff
@@ -129,7 +128,7 @@ export default function InventorySeedslingsTable(props: InventorySeedslingsTable
 
       if (activeRequests) {
         const batchesResults = searchResponse?.map((sr) => {
-          return { ...sr, facilityId: sr.facility_id } as Batch;
+          return { ...sr, facilityId: sr.facility_id };
```
Comment (karthikbtf) [inline src/components/Inventory/view/InventorySeedlingsTable.tsx:131]: search results are not `Batch` types from batch specific api. For example, totalQuantity does not exist in Batch but exists in search results

File: src/components/Inventory/withdraw/BatchWithdrawFlow.tsx [hunk 2]
```diff
@@ -94,6 +98,15 @@ export default function BatchWithdrawFlow(props: BatchWithdrawFlowProps): JSX.El
 
   const withdraw = async (photos: File[]) => {
     // first create the withdrawal
+    record.batchWithdrawals = record.batchWithdrawals.filter((batchWithdrawal) => {
+      return Number(batchWithdrawal.readyQuantityWithdrawn) + Number(batchWithdrawal.notReadyQuantityWithdrawn) > 0;
+    });
+
+    if (record.batchWithdrawals.length === 0) {
```
Comment (karthikbtf) [inline src/components/Inventory/withdraw/BatchWithdrawFlow.tsx:105]: BE cannot withdraw from entries if they are all 0, BE will return an error

**Review by karthikbtf — COMMENTED** [review]
some comments

File: src/components/Inventory/view/InventorySeedlingsTable.tsx [hunk 3]
```diff
@@ -232,17 +239,19 @@ export default function InventorySeedslingsTable(props: InventorySeedslingsTable
...
-          onUpdateOpenBatch(null);
-          setOpenNewBatchModal(false);
-        }}
-        organization={organization}
-        speciesId={speciesId}
-        selectedBatch={selectedBatch}
-      />
+      {openNewBatchModal && (
+        <BatchDetailsModal
+          open={openNewBatchModal}
```
Comment (charlie-tf) [inline src/components/Inventory/view/InventorySeedlingsTable.tsx:244]: Since `open={openNewBatchModal}` why conditionally render this?

Comment (karthikbtf) [reply src/components/Inventory/view/InventorySeedlingsTable.tsx:244]: We have some overloaded/complex modals that are hard to clean up on close. This conditional rendering allows for a clean instance every time we add a new batch. Not ideal.

Comment (karthikbtf) [reply src/components/Inventory/view/InventorySeedlingsTable.tsx]: fixed it

Comment (charlie-tf) [reply src/components/Inventory/view/InventorySeedlingsTable.tsx]: 🎉 thanks!

**Review by charlie-tf — APPROVED** [review]
looks good. Thanks!

**Comment by karthikbtf** [comment]
> looks good. Thanks!

Thanks Charlie!

**Pull request merged**